src/core_plugins/kibana/public/visualize/listing/visualize_landing_page_table.js

```diff
@@ -162,43 +118,30 @@ export class VisualizeLandingPageTable extends React.Component {
     return this.getColumnSortOrder(TYPE_COLUMN_ID);
   }
 
-  getAreAllItemsSelected() {
-    const { selectedIds, items, currentPage } = this.state;
-    const pageOfItems = this.pager.getItemsOnPage(items, currentPage);
-    const pageOfItemIds = pageOfItems.map(item => item.id);
-    return areAllItemsSelected(selectedIds, pageOfItemIds);
+  areAllItemsSelected() {
+    return areAllItemsSelected(this.state, this.pager);
   }
 
-  getVisualizeColumns() {
+  renderVisualizeColumns() {
     const { selectedIds } = this.state;
     return [
-      getCheckBoxColumn(this.getAreAllItemsSelected(), selectedIds, this.onToggleItem, this.onToggleAll),
-      getTitleColumn(this.getEditUrlForItem, this.getTitleSortOrder(), this.onSortByTitle),
-      getTypeColumn(this.onSortByType, this.getTypeSortOrder()),
+      getCheckBoxColumn(this.areAllItemsSelected(), selectedIds, this.onToggleItem, this.onToggleAll),
+      getTitleColumn(this.getEditUrlForItem, this.getTitleSortOrder(), this.onSort.bind(null, TITLE_COLUMN_ID)),
+      getTypeColumn(this.onSort.bind(null, TYPE_COLUMN_ID), this.getTypeSortOrder()),
     ];
   }
 
-  onDelete = () => {
-    const { deleteItems } = this.props;
-    const { selectedIds, filter } = this.state;
-    deleteItems(selectedIds).then((didDelete) => {
-      if (didDelete) {
-        this.onFilter(filter);
-      }
-    });
-  };
-
-  getActionButtons() {
+  renderActionButtons() {
     return this.state.selectedIds.length > 0
       ? <DeleteButton onClick={this.onDelete} />
       : <CreateButtonLink href={'#' + VisualizeConstants.WIZARD_STEP_1_PAGE_PATH} />;
   }
 
-  getTableContents() {
+  renderTableContents() {
```

Comment:
Some thoughts on designing Table components...

A Table is composed of two things: a TableHeader and a TableBody.

The TableHeader consists of a single row of cells. The TableBody consists of multiple rows, each of which consists of cells. So we can think about configuring both the TableHeader and TableBody with one-dimensional arrays (1 row of N cells).

Within these arrays, each index will contain a configuration for a particular cell (TableHeaderCell or TableRowCell). We'll need to be able to configure specific properties of each cell:

 - Cell content (children), an obvious one. This could literally be anything.
 - Presentational properties, e.g. whether overflowing content should be clipped. These could change over time as we discover new UI requirements.

If very similar instances of Tables exist in different parts of the UI, we could abstract out these configuration definitions into functions. We could call these functions with configuration parameters to customize specific parts of the table configurations for different contexts.

We could also abstract out the functions that render different parts of a Table, if that improves the code's readability.